report/4.Solution/LanuageOfCompiler.tex

```diff
@@ -0,0 +1,38 @@
+\section{The languages of the compiler}
+When creating a compiler there are three languages of relevance, as seen in the tombstone diagram\ref{fig:TStoneExample}.
+
+\begin{figure}[H]
+	\centering
+	\includegraphics[width=\textwidth/2+\textwidth/4]{4.Solution/images/T-diagram.png}
+	\caption{
+		Example of Tombstone diagram showcasing an example of three languages for a given compiler.\cite{TStoneWiki}
+	}
+	\label{fig:TStoneExample}
+\end{figure}
+In the case of the HCL compiler the three languages are as follows:
+
+\begin{align*}
+input \to & HCL \\
+output \to & C++/C \\
+compiler \to & Kotlin \\
+\end{align*}
+
+\subsection{Language of the compiler}
+The compiler was written in Kotlin\cite{KotlinWebsite}, a new programming language that compiles to both the JVM, javascript and LLVM. 
+The compiler is compiled to JVM because this grants access to all of the java ecosystem libraries, which in turn makes the JVM the most used target of kotlin.
+This choice to use kotlin was based on a number of factors.
+The group wanted to experiment with using a language they had no prior experience with for the duration of the project, but because of time limitations, a new language had to adhere to already known paradigms.
+The group knew C\# and a bit of Java, however as the HCL language has a lot of functional features, it was deemed important that the group became comfortable with a language that, at the very least was multi paradigm.
+Kotlin is inspired by both C\# and java and have a fairly similar syntax to that of those, which made it easier to learn from the perspective of the group.
+However it is still possible to write methods that is utilizes a functional domain.
+
+This grants the developer a large amount of possibilities, while still being relatively easy to learn.
+A disadvantage could be that low-level languages might have faster executing time, however as HCL is built with the Arduino architecture in mind, it is not possible to create vast programs, as the architecture has some hardware limitations, so execution time of the compiler will never really affect the end user, as the difference should not be noticeable.
```

Comment:
rebuild sentence
A disadvantage could be that low-level languages might have faster execution time.
However, as HCL is built with the Arduino architecture in mind, it is not possible to create vast programs due to hardware limitations.
Therefore, the execution time of the compiler will never really affect the end user, as the difference should not be noticeable